# IPL 2025 Player Analysis



In [15]:
import pandas as pd
import numpy as np

# Load the datasets
bat_df = pd.read_csv(r'IPL_2025\IPL2025Batters.csv')
bowl_df = pd.read_csv(r'IPL_2025\IPL2025Bowlers.csv')

# Preview Data
print(bat_df.head())
print(bowl_df.head())

         Player Name Team  Runs  Matches  Inn  No    HS    AVG   BF      SR  \
0      Sai Sudharsan   GT   759       15   15   1  108*  54.21  486  156.17   
1  Surya Kumar Yadav   MI   717       16   16   5   73*  65.18  427  167.91   
2        Virat Kohli  RCB   657       15   15   3   73*  54.75  454  144.71   
3       Shubman Gill   GT   650       15   15   2   93*  50.00  417  155.87   
4     Mitchell Marsh  LSG   627       13   13   0   117  48.23  383  163.70   

   100s  50s  4s  6s  
0     1    6  88  21  
1     0    5  69  38  
2     0    8  66  19  
3     0    6  62  24  
4     1    6  56  37  
       Player Name  Team  WKT  MAT  INN   OVR  RUNS   BBI    AVG   ECO     SR  \
0  Prasidh Krishna    GT   25   15   15  59.0   488  41/4  19.52  8.27  14.16   
1       Noor Ahmad   CSK   24   14   14  50.0   408  18/4  17.00  8.16  12.50   
2   Josh Hazlewood   RCB   22   12   12  44.0   386  33/4  17.54  8.77  12.00   
3      Trent Boult    MI   22   16   16  57.4   517  26/4  23.5

## Dataset Overview

In [16]:
# Display shapes and column names
print("Batting Data Shape:", bat_df.shape)
print("Bowling Data Shape:", bowl_df.shape)
print("Batting Columns:", bat_df.columns.tolist())
print("Bowling Columns:", bowl_df.columns.tolist())

Batting Data Shape: (156, 14)
Bowling Data Shape: (108, 13)
Batting Columns: ['Player Name', 'Team', 'Runs', 'Matches', 'Inn', 'No', 'HS', 'AVG', 'BF', 'SR', '100s', '50s', '4s', '6s']
Bowling Columns: ['Player Name', 'Team', 'WKT', 'MAT', 'INN', 'OVR', 'RUNS', 'BBI', 'AVG', 'ECO', 'SR', '4W', '5W']


In [17]:
# Check for missing values
print(bat_df.isnull().sum())
print(bowl_df.isnull().sum())

# Data types
print(bat_df.dtypes)
print(bowl_df.dtypes)

Player Name    0
Team           0
Runs           0
Matches        0
Inn            0
No             0
HS             0
AVG            0
BF             0
SR             0
100s           0
50s            0
4s             0
6s             0
dtype: int64
Player Name    0
Team           0
WKT            0
MAT            0
INN            0
OVR            0
RUNS           0
BBI            0
AVG            0
ECO            0
SR             0
4W             0
5W             0
dtype: int64
Player Name     object
Team            object
Runs             int64
Matches          int64
Inn              int64
No               int64
HS              object
AVG             object
BF               int64
SR             float64
100s             int64
50s              int64
4s               int64
6s               int64
dtype: object
Player Name     object
Team            object
WKT              int64
MAT              int64
INN              int64
OVR            float64
RUNS             int64
BBI             ob

## Batting Analysis

In [34]:
# Top 10 Run Scorers
top_batters = bat_df.sort_values(by='Runs', ascending=False).head(10)
top_batters.head()

Player Name Team  Runs  Matches  Inn  No    HS    AVG   BF      SR  \
0      Sai Sudharsan   GT   759       15   15   1  108*  54.21  486  156.17   
1  Surya Kumar Yadav   MI   717       16   16   5   73*  65.18  427  167.91   
2        Virat Kohli  RCB   657       15   15   3   73*  54.75  454  144.71   
3       Shubman Gill   GT   650       15   15   2   93*  50.00  417  155.87   
4     Mitchell Marsh  LSG   627       13   13   0   117  48.23  383  163.70   

   100s  50s  4s  6s  BoundaryRuns  
0     1    6  88  21           478  
1     0    5  69  38           504  
2     0    8  66  19           378  
3     0    6  62  24           392  
4     1    6  56  37           446

In [19]:
# Create BoundaryRuns column
bat_df['BoundaryRuns'] = bat_df['4s']*4 + bat_df['6s']*6
bat_df.sort_values(by='BoundaryRuns',ascending=False).head()

Player Name Team  Runs  Matches  Inn  No    HS    AVG   BF      SR  \
1   Surya Kumar Yadav   MI   717       16   16   5   73*  65.18  427  167.91   
0       Sai Sudharsan   GT   759       15   15   1  108*  54.21  486  156.17   
4      Mitchell Marsh  LSG   627       13   13   0   117  48.23  383  163.70   
10    Nicholas Pooran  LSG   524       14   14   2   87*  43.67  267  196.25   
6    Yashasvi Jaiswal   RR   559       14   14   1    75  43.00  350  159.71   

    100s  50s  4s  6s  BoundaryRuns  
1      0    5  69  38           504  
0      1    6  88  21           478  
4      1    6  56  37           446  
10     0    5  45  40           420  
6      0    6  60  28           408

In [20]:
# Statistical Calculations
mean_runs = np.mean(bat_df['Runs'])
median_runs = np.median(bat_df['Runs'])
percentiles = np.percentile(bat_df['Runs'], [25, 75])
std_runs = np.std(bat_df['Runs'])
mean_runs, median_runs, percentiles, std_runs

(161.32051282051282, 93.5, array([ 10.75, 254.25]), 179.77378902456803)

In [21]:
# High Strike Rate and 6s Filter
bat_df[(bat_df['SR'] > 150) & (bat_df['6s'] > 30)]

Player Name  Team  Runs  Matches  Inn  No   HS    AVG   BF      SR  \
1   Surya Kumar Yadav    MI   717       16   16   5  73*  65.18  427  167.91   
4      Mitchell Marsh   LSG   627       13   13   0  117  48.23  383  163.70   
5        Shreyas Iyer  PBKS   604       17   17   5  97*  50.33  345  175.07   
10    Nicholas Pooran   LSG   524       14   14   2  87*  43.67  267  196.25   

    100s  50s  4s  6s  BoundaryRuns  
1      0    5  69  38           504  
4      1    6  56  37           446  
5      0    6  43  39           406  
10     0    5  45  40           420

## Bowling Analysis

In [22]:
bowl_df.columns

Index(['Player Name', 'Team', 'WKT', 'MAT', 'INN', 'OVR', 'RUNS', 'BBI', 'AVG',
       'ECO', 'SR', '4W', '5W'],
      dtype='object')

In [23]:
# Top 10 Wicket Takers
top_bowlers = bowl_df.sort_values(by='WKT', ascending=False).head(10)
top_bowlers

Player Name  Team  WKT  MAT  INN   OVR  RUNS   BBI    AVG   ECO  \
0      Prasidh Krishna    GT   25   15   15  59.0   488  41/4  19.52  8.27   
1           Noor Ahmad   CSK   24   14   14  50.0   408  18/4  17.00  8.16   
2       Josh Hazlewood   RCB   22   12   12  44.0   386  33/4  17.54  8.77   
3          Trent Boult    MI   22   16   16  57.4   517  26/4  23.50  8.96   
4       Arshdeep Singh  PBKS   21   17   16  58.2   518  16/3  24.66  8.88   
5          Sai Kishore    GT   19   15   15  42.3   393  30/3  20.68  9.24   
6       Jasprit Bumrah    MI   18   12   12  47.2   316  22/4  17.55  6.67   
7  Varun Chakaravarthy   KKR   17   13   13  50.0   383  22/3  22.52  7.66   
8        Krunal Pandya   RCB   17   15   15  46.0   379  45/4  22.29  8.23   
9    Bhuvneshwar Kumar   RCB   17   14   14  52.0   483  33/3  28.41  9.28   

      SR  4W  5W  
0  14.16   1   0  
1  12.50   2   0  
2  12.00   1   0  
3  15.72   1   0  
4  16.66   0   0  
5  13.42   0   0  
6  15.77   1   0  
7  17.64   0   0  
8  16.23   1   0  
9  18.35   0   0

In [24]:
# Statistical Calculations
economy_mean = np.mean(bowl_df['ECO'])
wickets_max = np.max(bowl_df['WKT'])
bowling_avg_median = np.median(bowl_df['AVG'])
economy_mean, wickets_max, bowling_avg_median

(9.875092592592592, 25, 31.83)

In [25]:
# Filter Bowlers with Good Economy and Strike Rate
bowl_df[(bowl_df['ECO'] < 7) & (bowl_df['SR'] < 20)]

Player Name Team  WKT  MAT  INN   OVR  RUNS   BBI    AVG   ECO     SR  \
6  Jasprit Bumrah   MI   18   12   12  47.2   316  22/4  17.55  6.67  15.77   

   4W  5W  
6   1   0

## Combined Analysis

In [26]:
# Merge Batting and Bowling Data
combined_df = pd.merge(bat_df, bowl_df, on='Player Name', how='inner')
combined_df.head()

Player Name Team_x  Runs  Matches  Inn  No   HS  AVG_x   BF    SR_x  \
0    Aiden Markram    LSG   445       13   13   0   66  34.23  299  148.82   
1      Riyan Parag     RR   393       14   14   2   95  32.75  236  166.52   
2     Ayush Badoni    LSG   329       14   11   1   74  32.90  222  148.19   
3  Ravindra Jadeja    CSK   301       14   14   5  77*  33.44  222  135.58   
4       Axar Patel     DC   263       12   11   1   43  26.30  167  157.48   

   ...  MAT  INN   OVR  RUNS    BBI  AVG_y   ECO  SR_y  4W  5W  
0  ...   13    5  11.0   102   30/2  25.50  9.27  16.5   0   0  
1  ...   14    9  20.0   170  1-Dec  56.66  8.50  40.0   0   0  
2  ...   14    2   1.4    13  2-Apr   6.50  7.80   5.0   0   0  
3  ...   14   14  37.5   324   17/2  32.40  8.56  22.7   0   0  
4  ...   12   11  34.0   288   19/2  57.60  8.47  40.8   0   0  

[5 rows x 27 columns]

In [27]:
# Identify All-rounders
all_rounders = combined_df[(combined_df['Runs'] > 200) & (combined_df['WKT'] > 10)]
all_rounders[['Player Name','Runs','WKT']]

Player Name  Runs  WKT
5   Sunil Narine   246   12
7  Hardik Pandya   224   14

In [28]:
# All-rounder Score
max_runs = combined_df['Runs'].max()
max_wickets = combined_df['WKT'].max()
combined_df['Score'] = (combined_df['Runs'] / max_runs) * 0.5 + (combined_df['WKT'] / max_wickets) * 0.5
combined_df.sort_values(by='Score', ascending=False).head(5)

Player Name Team_x  Runs  Matches  Inn  No   HS  AVG_x   BF    SR_x  \
0     Aiden Markram    LSG   445       13   13   0   66  34.23  299  148.82   
3   Ravindra Jadeja    CSK   301       14   14   5  77*  33.44  222  135.58   
7     Hardik Pandya     MI   224       15   12   3  48*  24.89  137  163.50   
5      Sunil Narine    KKR   246       12   12   1   44  22.36  144  170.83   
49       Noor Ahmad    CSK     7       14    6   2   2*   1.75   17   41.17   

    ...  INN   OVR  RUNS   BBI  AVG_y   ECO  SR_y  4W  5W     Score  
0   ...    5  11.0   102  30/2  25.50  9.27  16.5   0   0  0.583333  
3   ...   14  37.5   324  17/2  32.40  8.56  22.7   0   0  0.546536  
7   ...   14  35.0   342  36/5  24.42  9.77  15.0   0   1  0.543352  
5   ...   12  45.0   351  13/3  29.25  7.80  22.5   0   0  0.526404  
49  ...   14  50.0   408  18/4  17.00  8.16  12.5   2   0  0.507865  

[5 rows x 28 columns]

Insights
1. Best All rounder players are Hardik Pandya and Sunil Narine
2. Best Batsman Sai Sudarshan 759 out of 15 matches with strike rate of 167.9
3. Best Bowler Prasidh Krishna took 25 wickets with Economy 8.27